### ISSUE TITLE
Questions about the impact of image modality on model performance

### ISSUE THREAD

**Issue by dongweiLiu** [CityMind-Lab/ICML25-TimeVLM#14]
Dear author, thank you for your excellent work. After modifying the class **LearnableTimeSeriesToImage** in **Learnable_TimeSeries_To_Image.py** to randomly generate images as input to CLIP and retraining the model. I observed only a minimal change in performance (approximately 1%):
```
class LearnableTimeSeriesToImage(nn.Module):
    """Learnable module to convert time series data into image tensors"""

    def __init__(self, image_size=56):
        super(LearnableTimeSeriesToImage, self).__init__()
        self.image_size = image_size

    def forward(self, x_enc):
        """Convert input time series to image tensor [B, output_channels, H, W]"""
        B, L, D = x_enc.shape
        return torch.rand((B, 3, self.image_size, self.image_size))  # [B, output_channels, H, W]
```

<img width="774" height="744" alt="Image" src="https://github.com/user-attachments/assets/834149c4-2c4e-4d46-94a2-65973b372439" />

Furthermore, while debugging the original model pretrained on the Weather dataset (with pre_len set to 96), I found that the variable **gate_weights** assigned a weight of 0.99 to temporal features and only 0.01 to multimodal features:

<img width="1357" height="695" alt="Image" src="https://github.com/user-attachments/assets/375c00a1-a9c7-4df7-acfb-b16bb85d471e" />

I would like to ask if your experimental results are consistent with the above phenomena? Does the above phenomenon indicate that the influence of image modal branching on model performance is actually very small? I sincerely appreciate your insights on these questions.

**Comment by siruzhong** [comment]
Your experiment is interesting. Actually, we recently did another work on what kinds of images are useful in this time-series to image setting. You can refer to OccamVTS [1] if you're interested. Through cross-modal analysis, we found that time series align with low-level textural features rather than high-level semantics, which can impair forecasting accuracy. The random images you generated may look similar to this kind of images in some datasets. From my perspective, I think images just provide an embedding space for multimodal knowledge. As long as the generated images are in line with nature and, in terms of spatial distribution, most similar to a time series, then they are useful.

Regarding the gate values, the gate distribution varies significantly across datasets. Maybe you can try other datasets like ETT or Traffic, or you can gradually observe the values change. Generally, they start high (0.6/0.5/0.4) in the early epochs and gradually drop to around a fixed value (this varies by dataset - some can have 0.2, some may only reach 0.1 or 0.05). It's a normal phenomenon, as multimodal information can provide some information when data is limited. When there's sufficient data, we can get enough information from the time series itself. Time-VLM's key insight is that the time series itself is most important, while multimodal features can only provide auxiliary information.

Visual interpretation of time series is also in the early stages of research. Maybe some listed works are useful for you [2][3].

Besides, ETT, Traffic, and Weather are really small and outdated datasets. We believe this is a future research direction that can be applied to more modern and larger datasets.

[1] OccamVTS: Distilling Vision Models to 1% Parameters for Time Series Forecasting

[2] From Images to Signals: Are Large Vision Models Useful for Time Series Analysis?

[3] Harnessing Vision Models for Time Series Analysis: A Survey

**Comment by dongweiLiu** [comment]
Thank for your reply. I will read these papers.
